### PR TITLE
[chore] [exporter/datadog] Refactor host metadata in terms of a pusher

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

Refactors current host metadata logic in preparation for using `inframetadata.Reporter`.

This is an internal-only change with no user-facing consequences

### CODE REVIEW

File: exporter/datadogexporter/internal/hostmetadata/metadata.go [hunk 1]
```diff
@@ -150,21 +164,25 @@ func Pusher(ctx context.Context, params exporter.CreateSettings, pcfg PusherConf
...
+	hostMetadata := payload.HostMetadata{Meta: &payload.Meta{}, Tags: &payload.HostTags{}}
 	if pcfg.UseResourceMetadata {
 		hostMetadata = metadataFromAttributes(attrs)
 	}
-	fillHostMetadata(params, pcfg, p, hostMetadata)
+	fillHostMetadata(params, pcfg, p, &hostMetadata)
 
 	// Run one first time at startup
-	pushMetadataWithRetry(retrier, params, pcfg, hostMetadata)
+	pusher.Push(ctx, hostMetadata)
```
Comment (songy23) [inline exporter/datadogexporter/internal/hostmetadata/metadata.go:174]: Do you want to check the return error here?

File: exporter/datadogexporter/internal/hostmetadata/metadata.go [hunk 2]
```diff
@@ -120,28 +113,49 @@ func pushMetadata(pcfg PusherConfig, params exporter.CreateSettings, metadata *p
...
+var _ inframetadata.Pusher = (*pusher)(nil)
+
+type pusher struct {
+	params     exporter.CreateSettings
+	pcfg       PusherConfig
+	retrier    *clientutil.Retrier
+	httpClient *http.Client
+}
+
+// NewPusher creates a new inframetadata.Pusher that pushes
```
Comment (songy23) [inline exporter/datadogexporter/internal/hostmetadata/metadata.go:141]:
```suggestion
// NewPusher creates a new inframetadata.Pusher that pushes metadata payloads
```

**Review by songy23 — COMMENTED** [review]
Needs `make gotidy`

File: exporter/datadogexporter/internal/hostmetadata/metadata.go (hunk 1 above)
Comment (mx-psi) [reply exporter/datadogexporter/internal/hostmetadata/metadata.go]: Good point, checking it now

**Review by songy23 — APPROVED** [review]
LGTM

**Pull request merged**